### ISSUE TITLE
[BUG] sigmoid hidden activation in CNN network, classifier & regressor

### ISSUE THREAD

**Issue by RecreationalMath** [sktime/sktime#9049]
**Describe the bug**

CNN Network, classifier & regressor currently has `sigmoid` as default hidden layer activation.

---

**To Reproduce**

Check [line 53](https://github.com/sktime/sktime/blob/main/sktime/networks/cnn.py#L53) in `CNNNetwork` class.
Check [line 110](https://github.com/sktime/sktime/blob/main/sktime/regression/deep_learning/cnn.py#L110) in `CNNRegressor` class.
Check [line 103](https://github.com/sktime/sktime/blob/main/sktime/classification/deep_learning/cnn.py#L103) in `CNNClassifier` class.

---

**Expected behaviour & Additional context**

In the hidden layers of a CNN, it is non-standard to use `sigmoid` as an activation. Sigmoid outputs are in (0,1) and not zero-centred, which can slow convergence. Also, in general sigmoid is used a final layer activation in binary classification. 

`ReLU` or a variation of ReLU is a generally agreeable choice for activation in hidden layers of a CNN.

---

**Versions**
<details>

System:
    python: 3.12.10 (main, Oct 30 2025, 20:57:26) [Clang 17.0.0 (clang-1700.3.19.1)]
executable: /.pyenv/versions/sktime-dev-py-3.12.10/bin/python
   machine: macOS-15.6.1-arm64-arm-64bit

Python dependencies:
          pip: 25.3
       sktime: 0.39.0
      sklearn: 1.6.1
       skbase: 0.12.6
        numpy: 2.1.3
        scipy: 1.15.3
       pandas: 2.3.3
   matplotlib: 3.10.7
       joblib: 1.5.2
huggingface-hub: None
        numba: 0.62.1
     pmdarima: None
pytorch-forecasting: None
   skforecast: 0.18.0
        skpro: 2.9.4
statsforecast: 2.0.3
  statsmodels: 0.14.5
 transformers: None
      tsfresh: None
      tslearn: None
        torch: 2.9.0
   tensorflow: 2.19.1

</details>
